sql/src/main/java/io/crate/metadata/cluster/AlterTableClusterStateExecutor.java

```diff
@@ -0,0 +1,437 @@
+                if (indexScopedSettings.updateSettings(closedSettings, indexSettings, updates, index.getName())) {
+                    //if (preserveExisting) {
+                    //  indexSettings.put(indexMetaData.getSettings());
+                    //}
+                    Settings finalSettings = indexSettings.build();
+                    indexScopedSettings.validate(finalSettings.filter(k -> indexScopedSettings.isPrivateSetting(k) ==
+                                                                           false), true);
+                    metaDataBuilder.put(IndexMetaData.builder(indexMetaData).settings(finalSettings));
+                }
+            }
+        }
+
+        // increment settings versions
+        for (final String index : actualIndices) {
+            if (same(currentState.metaData().index(index).getSettings(), metaDataBuilder.get(index).getSettings()) ==
+                false) {
+                final IndexMetaData.Builder builder = IndexMetaData.builder(metaDataBuilder.get(index));
+                builder.settingsVersion(1 + builder.settingsVersion());
+                metaDataBuilder.put(builder);
+            }
+        }
+
+        ClusterState updatedState = ClusterState.builder(currentState).metaData(metaDataBuilder).routingTable(
+            routingTableBuilder.build()).blocks(blocks).build();
+
+        //TODO do we still need the following operations and are they atomic ?
+        // now, reroute in case things change that require it (like number of replicas)
+        updatedState = allocationService.reroute(updatedState, "settings update");
+        try {
+            for (Index index : openIndices) {
+                final IndexMetaData currentMetaData = currentState.getMetaData().getIndexSafe(index);
+                final IndexMetaData updatedMetaData = updatedState.metaData().getIndexSafe(index);
+                indicesService.verifyIndexMetadata(currentMetaData, updatedMetaData);
+            }
+            for (Index index : closeIndices) {
+                final IndexMetaData currentMetaData = currentState.getMetaData().getIndexSafe(index);
+                final IndexMetaData updatedMetaData = updatedState.metaData().getIndexSafe(index);
+                // Verifies that the current index settings can be updated with the updated dynamic settings.
+                indicesService.verifyIndexMetadata(currentMetaData, updatedMetaData);
+                // Now check that we can create the index with the updated settings (dynamic and non-dynamic).
+                // This step is mandatory since we allow to update non-dynamic settings on closed indices.
+                indicesService.verifyIndexMetadata(updatedMetaData, updatedMetaData);
+            }
+        } catch (IOException ex) {
+            throw ExceptionsHelper.convertToElastic(ex);
+        }
+        return updatedState;
+    }
+
+    private static void maybeUpdateClusterBlock(String[] actualIndices,
```

Comment:
can't we use `MetaDataUpdateSettingsService.maybeUpdateClusterBlock()`  instead?